nvflare/app_common/np/recipes/lr/fedavg.py

```diff
@@ -0,0 +1,137 @@
+# Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, List, Optional
+
+from pydantic import BaseModel, PositiveInt
+
+from nvflare import FedJob
+from nvflare.apis.analytix import ANALYTIC_EVENT_TYPE
+from nvflare.app_common.widgets.convert_to_fed_event import ConvertToFedEvent
+from nvflare.app_common.workflows.lr.fedavg import FedAvgLR
+from nvflare.app_common.workflows.lr.np_persistor import LRModelPersistor
+from nvflare.app_opt.tracking.tb.tb_receiver import TBAnalyticsReceiver
+from nvflare.client.config import ExchangeFormat, TransferType
+from nvflare.job_config.script_runner import ScriptRunner, FrameworkType
+from nvflare.recipe.spec import Recipe
+
+
+# Internal — not part of the public API
+class _FedAvgValidator(BaseModel):
+    name: str
+    initial_model: Any
+    clients: Optional[List[str]]
+    num_clients: Optional[PositiveInt]
+    min_clients: int
+    num_rounds: int
+    damping_factor: float
+    train_script: str
+    train_args: str
+    launch_external_process: bool = False
+    command: str
+
+    def model_post_init(self, __context):
+        if self.clients and self.num_clients is None:
+            self.num_clients = len(self.clients)
+        elif self.clients and len(self.clients) != self.min_clients:
+            raise ValueError("inconsistent number of clients")
+
+
+class FedAvgLrRecipe(Recipe):
+    def __init__(
+            self,
+            *,
+            name: str = "lr_fedavg",
+            initial_model: Any = None,
+            clients: Optional[List[str]] = None,
+            num_clients: Optional[int] = None,
```

Comment:
might need to add the docstring to explain these a bit, we can just copy from FedAvgRecipe